intro-to-oop-contd/projects-with-many-files.md

```diff
@@ -147,40 +171,176 @@ We must _import_ any resources that we need from another module before we can us
+
+```python
+one = UsefulClassOne()
+one.do_something()
+two = UsefulClassTwo()
+two.do_something_else()
+```
+
+### So Many Ways to Import
+
+As we have shown, there are _many_ different syntaxes and strategies we can use to import modules. If a particular import syntax works for a project, it's valid!
+
+Different situations will need to use different ways of importing. So we should be ready to see many styles of importing. We may even encounter styles that we haven't discussed that we might need to look up. That's ok! But we should feel confident that we can recognize importing when we see it!
+
+## Restructuring a Project
+
+Let's return to Scarlet's project and see how she decides to reorganize her project. Remember that she started with a single file `main.py` that contained two classes, `Driver` and `Passenger`, and there was some main logic to create a `Passenger` instance.
+
+Scarlet is making a ride share app, so she decides to call her project root `ride-share-app`. She puts her standard project files in the project root. She knows that in Python we like to have each class in its own module, and that the modules for a project are usually grouped into a package. She can't use hyphens in the package name, so she decides on `ride_share_app` as her package name. She makes the folder, adds the `__init__.py` file to tell Python this is a package folder, then makes a module file for each class: `driver.py` and `passenger.py`.
+
+She considers renaming the `main.py` which now only contains the application startup to `ride_share.py` to be a little more descriptive on the command line. But Scarlet learned about another strategy for where to put her startup code that she wants to try, so for now she leaves the code in `main.py`
+
+### !callout-warning
+
+## Be Careful With Descriptive Entry Point File Names
+
+Scarlet also briefly considered renaming `main.py` to `ride_share_app.py`, but then she realized that would prevent Python from being able to see the `ride_share_app` package. If the main file has the same name as the package, if we try to import from the package, Python will try looking inside the module we are currently in, not find the desired identifier, and raise an `ImportError`.
+
+<br />
+
+We should make sure that the entry point file name does _not_ match the project package name. Alternatives include adding `_pkg` to the name of the package, though this isn't very Pythonic. Or we could make the entry point closely related to, but not the same as the package name, like how Scarlet considered `ride_share.py`. We can use a more general entry point, like `main.py`. Or if we want to be able to use the package name to launch the application, we can use the special `__main__.py` file described later in this lesson.
+
+### !end-callout
```

Comment:
The students don't necessarily have meaningful context or examples to have run into this, felt this pain, or the motivation to do something with this knowledge, IMO.

I think it personally takes until capstone to begin to have opinions on file names.

Also, the concept of an "entry point" is pretty intangible and far to them IMO. The closest entry point that they may see may be a Flask project depending on how you think about it. Even at that point, I think it's a lot of explanation to think about an entry point and it'd get awkward without the proper way to visualize it.

In that whole theme, I think that the idea of Scarlet considering renaming `main.py` is a really out-of-reach consideration. I don't think the knowledge/scenario presented here will be the juiciest for students.

Personally, I'm into the idea of cutting it, to make room for the other good content in here?

```suggestion
```

Reply:
You already heard the back story for this. But as we're going to deemphasize the entry point stuff, this can go!